### PR TITLE
Add Address Sanitizer Support

### PR DESCRIPTION
**Pull request by chiragramani**
This PR adds Address Sanitizer Support.

- Add the following example in the sanitizers app and enable Address Sanitizer.
```
let pointer = UnsafeMutableRawPointer.allocate(byteCount: 1, alignment: 1)
pointer.storeBytes(of: 1, as: UInt8.self)
pointer.advanced(by: 1).storeBytes(of: 2, as: UInt8.self)
```

- Since we are advancing a pointer to take care of 2 bytes but the declared byteCount is 1, it should give a Heap Overflow Asan error.
- Tested and it works as expected.

TODO:

- AddressSanitizer application will be added inside `examples/sanitizers` similar to Tsan in the next PR.

### CODE REVIEW

File: tools/generator/src/Extensions/XCScheme+Extensions.swift [hunk 1]
```diff
@@ -171,7 +171,7 @@ echo "$BAZEL_LABEL,$BAZEL_TARGET_ID" >> "$SCHEME_TARGET_IDS_FILE"
     ) -> XCScheme.ExecutionAction {
         return .init(
             scriptText: #"""
-if [ "${ENABLE_THREAD_SANITIZER:-}" == "YES" ]; then
+if [ "${ENABLE_THREAD_SANITIZER:-}" == "YES" ] || [ "${ENABLE_ADDRESS_SANITIZER:-}" == "YES" ]; then
```
Comment (brentleyjones) [inline tools/generator/src/Extensions/XCScheme+Extensions.swift:174]:
```suggestion
if [ "${ENABLE_ADDRESS_SANITIZER:-}" == "YES" ] || [ "${ENABLE_THREAD_SANITIZER:-}" == "YES" ]; then
```

File: xcodeproj/internal/bazel_integration_files/bazel_build.sh [hunk 2]
```diff
@@ -195,6 +195,8 @@ elif [ "${ENABLE_PREVIEWS:-}" == "YES" ]; then
   config="${BAZEL_CONFIG}_swiftuipreviews"
 elif [ "${ENABLE_THREAD_SANITIZER:-}" == "YES" ]; then
   config="${BAZEL_CONFIG}_tsan"
+elif [ "${ENABLE_ADDRESS_SANITIZER:-}" == "YES" ]; then
```
Comment (brentleyjones) [inline xcodeproj/internal/bazel_integration_files/bazel_build.sh:198]: Can you put this above the thread sanitizer?

File: xcodeproj/internal/xcodeproj.template.bazelrc [hunk 3]
```diff
@@ -99,6 +99,13 @@ build:rules_xcodeproj_swiftuipreviews --features=-swift.index_while_building
 build:rules_xcodeproj_tsan --config=rules_xcodeproj
 build:rules_xcodeproj_tsan --features=tsan
 
+### `--config=rules_xcodeproj_asan`
```
Comment (brentleyjones) [inline xcodeproj/internal/xcodeproj.template.bazelrc:102]: Can you put this above the thread sanitizer?

File: xcodeproj/internal/xcodeproj_runner.bzl [hunk 4]
```diff
@@ -30,6 +30,8 @@ build:{config}_swiftuipreviews --config=rules_xcodeproj_swiftuipreviews
 build:{config}_swiftuipreviews --config={config}
 build:{config}_tsan --config=rules_xcodeproj_tsan
 build:{config}_tsan --config={config}
+build:{config}_asan --config=rules_xcodeproj_asan
```
Comment (brentleyjones) [inline xcodeproj/internal/xcodeproj_runner.bzl:33]: Can you put this above the thread sanitizer?

**Review by brentleyjones — APPROVED** [review]

File: xcodeproj/internal/xcodeproj.template.bazelrc (hunk 3 above)
Comment (chiragramani) [reply xcodeproj/internal/xcodeproj.template.bazelrc:95]: 👍🏼

File: xcodeproj/internal/xcodeproj_runner.bzl (hunk 4 above)
Comment (chiragramani) [reply xcodeproj/internal/xcodeproj_runner.bzl:31]: Done, thanks!

**Review by brentleyjones — APPROVED** [review]

**Pull request merged**